src/components/SplitPane.jsx

```diff
@@ -0,0 +1,146 @@
+var React = require('react'),
+    classNames = require('classnames'),
+    Pane = require('./Pane.jsx'),
+    Resizer = require('./Resizer.jsx'),
+    style = require('./component-playground.less');
+
+export default React.createClass({
+  displayName: 'SplitPane',
+
+  propTypes: {
+    minSize: React.PropTypes.number,
+    defaultSize: React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.number,
+    ]),
+    split: React.PropTypes.string,
+    onChange: React.PropTypes.func
+  },
+
+  getInitialState: function() {
+    return {
+      active: false,
+      resized: false
+    };
+  },
+
+  getDefaultProps: function() {
+    return {
+      minSize: 0
+    };
+  },
+
+  componentDidMount: function() {
+    document.addEventListener('mouseup', this.onMouseUp);
+    document.addEventListener('mousemove', this.onMouseMove);
+    var ref = this.refs.pane1;
+    if (ref && this.props.defaultSize && !this.state.resized) {
+      this.setState({
```

Comment:
Each component should manage its state, calling setState directly on a child is a React no-go. Can we instead store the size in SplitPane state and pass it as a prop to Pane ?